### PR TITLE
making change to capture inference endpoint for openai

### PR DESCRIPTION
**Pull request by Hansrajr**
## Proposed changes

This PR capturing inference endpoint for openai ( langchain, llama_index)

## Types of changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/monocle2ai/monocle/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/monocle_apptrace/wrap_common.py [hunk 1]
```diff
@@ -291,10 +291,11 @@ def update_llm_endpoint(curr_span: Span, instance):
 
 def set_provider_name(instance):
     provider_url = ""
-
+    inference_endpoint = ""
     try:
         if isinstance(instance.client._client.base_url.host, str):
             provider_url = instance.client._client.base_url.host
+            inference_endpoint = str(instance.client._client.base_url)
```
Comment (oi-raanne) [inline src/monocle_apptrace/wrap_common.py:298]: why str is needed ?

File: src/monocle_apptrace/wrap_common.py [hunk 2]
```diff
@@ -291,10 +291,11 @@ def update_llm_endpoint(curr_span: Span, instance):
 
 def set_provider_name(instance):
```
Comment (kshitiz-okahu) [inline src/monocle_apptrace/wrap_common.py:292]: We should rename this to get_provider_names .

File: src/monocle_apptrace/wrap_common.py (hunk 1 above)
Comment (Hansrajr) [reply src/monocle_apptrace/wrap_common.py:298]: The base_url is an instance of the URL class, so using str() will providefull URL as a plain string.

File: src/monocle_apptrace/wrap_common.py (hunk 2 above)
Comment (Hansrajr) [reply src/monocle_apptrace/wrap_common.py]: done

**Pull request merged**